thrust/thrust/iterator/zip_iterator.h

```diff
@@ -192,7 +195,19 @@ class _CCCL_DECLSPEC_EMPTY_BASES zip_iterator : public detail::make_zip_iterator
   //!
   //! \param iterator_tuple The \p tuple of iterators to copy from.
   inline _CCCL_HOST_DEVICE zip_iterator(IteratorTuple iterator_tuple)
-      : m_iterator_tuple(iterator_tuple)
+      : m_iterator_tuple(::cuda::std::move(iterator_tuple))
+  {}
+
+  //! This constructor creates a new \p zip_iterator from a pack of iterators.
+  //!
+  //! \param iterators A pack of iterators to zip.
+  template <
+    typename... Iterators,
+    ::cuda::std::enable_if_t<(::cuda::std::input_or_output_iterator<::cuda::std::remove_reference_t<Iterators>> && ...)
+                               && ::cuda::std::is_constructible_v<IteratorTuple, Iterators&&...>,
+                             int> = 0>
+  _CCCL_HOST_DEVICE zip_iterator(Iterators&&... iterators)
```

Comment:
@miscco is the `&&` in `::cuda::std::is_constructible_v<IteratorTuple, Iterators&&...>` correct? It is what `forward<Iterators>(...)` returns.

Reply:
It should just be `::cuda::std::is_constructible_v<IteratorTuple, Iterators...>`

Reply:
Fixed